packages/compass-sidebar/src/components/csfle-marker/csfle-marker.jsx

```diff
@@ -8,23 +8,38 @@ const badgeContainerStyles = css({
   padding: '0px 0px 6px 36px'
 });
 
-function CSFLEMarker({ isCSFLEConnection }) {
-  if (!isCSFLEConnection) {
+const badgeButtonStyles = css({
+  '&': {
+    background: 'inherit',
+    padding: 0,
+    margin: 0,
+    border: 'none',
+  },
+  '&:hover': {
+    cursor: 'pointer'
+  }
+});
+
+function CSFLEMarker({ csfleMode, toggleCSFLEModalVisible }) {
+  if (!csfleMode || csfleMode === 'unavailable') {
     return null;
   }
   return (
     <div className={badgeContainerStyles}>
-      <Badge variant={BadgeVariant.DarkGray}>
-        <Icon glyph="Key" />
-        CSFLE
-      </Badge>
+      <button className={badgeButtonStyles} onClick={() => toggleCSFLEModalVisible()}>
```

Comment:
I think for this button we probably want to `useFocusRing` and include an `aria-label` to improve accessibility and make the interactions consistent with some of the other newer buttons in Compass.
Here's an example of a place we're doing this already:
https://github.com/mongodb-js/compass/blob/47e5bfae7ee7b3424901c01bf66da21a0a9096e4/packages/compass-sidebar/src/components/sidebar/navigation-items.jsx#L63

Reply:
Sure, done!